### PR TITLE
Generate Jekyll and Doxygen for deployement

### PR DESCRIPTION
**Pull request by mwestphal**
- Add a doxygen generation mechanism
- Improve jekyll website and add links to doxygen
- Add a CI to generate both and deploy to gh-pages
- Add doc
- Result can be consulted here (before  URL update): https://mwestphal.github.io/f3d/

### CODE REVIEW

File: .github/workflows/generate-website.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      uses: actions/checkout@v3
+
+    - name: Build Jekyll
+      uses: actions/jekyll-build-pages@v1
+
+    - name: Install exhale and doxygen
+      working-directory: ${{github.workspace}}
+      run: |
+        pip install sphinx sphinx-book-theme exhale
+        sudo apt-get install doxygen
```
Comment (Meakk) [inline .github/workflows/generate-website.yml:27]: do you need sudo? I think you are already root here.

File: .github/workflows/generate-website.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    - name: Build doxygen
+      working-directory: ${{github.workspace}}/library/doxygen
+      run: make html
+
+    - name: Copy doxygen
```
Comment (Meakk) [inline .github/workflows/generate-website.yml:33]: Can you merge this step with the previous one? The `Makefile` is ridiculously simple, I would remove it completely and call the command with the correct build directory directly.

File: library/doxygen/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    # Suggested optional arguments
+    "createTreeView":        True,
+    # TIP: if using the sphinx-bootstrap-theme, you need
+    # "treeViewIsBootstrap": True,
+    "exhaleExecutesDoxygen": True,
+    "exhaleDoxygenStdin":    "INPUT = ../../public\nDISTRIBUTE_GROUP_DOC = YES\nEXCLUDE_SYMBOLS = std"
+}
+
+# Tell sphinx what the primary language being documented is.
+primary_domain = 'cpp'
```
Comment (Meakk) [inline library/doxygen/source/conf.py:82]: can we document the CMake files too? `f3dEmbed.cmake` is already in the SDK, and `f3dPlugin.cmake` will be included soon too.

File: library/doxygen/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+   contain the root `toctree` directive.
+
+libf3d doxygen documentation
+============================
+
+This is the automatically generated doxygen documentation for the libf3d_,
+which is part of F3D_.
+
+.. _libf3d: https://f3d.app/doc/libf3d/OVERVIEW.html
+.. _F3D: https://f3d.app/doc/libf3d/OVERVIEW.html
```
Comment (Meakk) [inline library/doxygen/source/index.rst:13]: Incorrect link?

File: library/doxygen/source/conf.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "rootFileName":          "library_root.rst",
+    "doxygenStripFromPath":  "..",
+    # Heavily encouraged optional argument (see docs)
+    "rootFileTitle":         "Library API",
+    # Suggested optional arguments
+    "createTreeView":        True,
+    # TIP: if using the sphinx-bootstrap-theme, you need
+    # "treeViewIsBootstrap": True,
+    "exhaleExecutesDoxygen": True,
+    "exhaleDoxygenStdin":    "INPUT = ../../public\nDISTRIBUTE_GROUP_DOC = YES\nEXCLUDE_SYMBOLS = std"
```
Comment (Meakk) [inline library/doxygen/source/conf.py:78]: If you remove the makefiles, I suggest to move `conf.py` and `index.rst` in `library/doxygen` directly and remove one level of `..` here.

**Review by Meakk — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-website.yml (hunk 1 above)
Comment (mwestphal) [reply .github/workflows/generate-website.yml:27]: nop, I'm not root, but the sudo is passwordless.

File: library/doxygen/source/conf.py (hunk 3 above)
Comment (mwestphal) [reply library/doxygen/source/conf.py:82]: no idea if doxygen supports that, ill take a look.

File: library/doxygen/source/index.rst (hunk 4 above)
Comment (mwestphal) [reply library/doxygen/source/index.rst:13]: indeed

File: library/doxygen/source/conf.py (hunk 5 above)
Comment (mwestphal) [reply library/doxygen/source/conf.py:78]: why would I remove the makefile ? I cant build the doc without it.

Comment (mwestphal) [reply library/doxygen/source/conf.py:78]: Ok, I understand your previous comment now. I'll try it.

File: library/doxygen/source/conf.py (hunk 3 above)
Comment (mwestphal) [reply library/doxygen/source/conf.py:82]: Doxygen do not support documenting cmake files: 

https://www.doxygen.nl/manual/starting.html#default_file_extension_mapping

I tried to force parsing on f3dPlugin.cmake anyway and it fails with:

```
/home/glow/dev/f3d/f3d/src/library/doxygen/api/program_listing_file__home_glow_dev_f3d_f3d_src_cmake_f3dPlugin.h.rst:11: WARNING: Could not lex literal_block as "cpp". Highlighting skipped.
```

**Review by Meakk — APPROVED** [review]

**Pull request merged**